### PR TITLE
frame-support-test: migrate tests from `decl_*` macros to the new `pallet` macros

### PR DESCRIPTION
**Pull request by koushiro**
## What does it do?

This PR is a part of #12248 and it need #12401 to be merged firstly

In this PR, I migrate the tests of `frame/support/test` from  old `decl_*` macros to the new `pallet` attribute macros, and remove some useless tests.

After this, I will remove the `decl_*` macros and do some other trivial things.

### CODE REVIEW

**Comment by koushiro** [comment]
I plan to use `frame-system` directly as base config to replace `frame-support-test` pallet and `system` module. How about it?

**Review by kianenigma — APPROVED** [review]
Didn't look too closely, but LGTM.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2056703

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2729586

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2730162

File: frame/support/test/tests/issue2219.rs [hunk 1]
```diff
@@ -83,102 +79,108 @@ mod module {
...
+	pub type Bondage<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, T::BlockNumber>;
+
+	/// First step before enter a role is registering intent with a new account/key.
+	/// This is done by sending a role_entry_request() from the new account.
+	/// The member must then send a stake() transaction to approve the request and enter the desired
+	/// role. The account making the request will be bonded and must have
+	/// sufficient balance to cover the minimum stake for the role.
+	/// Bonding only occurs after successful entry into a role.
+	#[pallet::storage]
+	#[pallet::getter(fn role_entry_requests)]
```
Comment (ggwpez) [inline frame/support/test/tests/issue2219.rs:127]: I will keep the getters in the tests as we should remove them here last IMHO as to not accidentally break them.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2750320

**Comment by ggwpez** [comment]
bot rebase

**Comment by ggwpez** [comment]
bot rebase

**Review by ggwpez — APPROVED** [review]

**Comment by ggwpez** [comment]
bot merge

**Comment by sam0x17** [comment]
showerthought: would be cool if they could just specify it in their github bio 🤔

**Pull request merged**